sdk-node/src/workflows/workflow.ts

```diff
@@ -277,18 +282,39 @@ export class Workflow<TInput extends WorkflowInput, name extends string> {
       executionId,
     });
 
-    const l1m = new L1M({
-      baseUrl: `${this.endpoint}/clusters/${clusterId}/l1m`,
-      additionalHeaders: {
-        "x-workflow-execution-id": executionId,
-        Authorization: `Bearer ${this.apiSecret}`,
-      },
-      provider: {
-        model: "claude-3-5-sonnet",
-        key: "",
-        url: "",
+    const llm = {
+      structured: async <T>(params: {
+        input: string;
+        instructions?: string;
+        schema: z.ZodType<T>;
+      }): Promise<T> => {
+        const result = await this.client.l1mStructured({
+          params: {
+            clusterId,
+          },
+          body: {
+            input: params.input,
+            instructions: params.instructions,
+            schema: zodToJsonSchema(params.schema),
+          },
+          headers: {
+            authorization: `Bearer ${this.apiSecret}`,
+            "x-provider-model": "claude-3-5-sonnet",
+            "x-provider-url": "",
+            "x-provider-key": "",
+            "x-workflow-execution-id": executionId,
+          },
+        });
+
+        if (result.status !== 200) {
+          throw new Error(
+            `L1M structured call failed with status ${result.status}`,
```

Comment:
L1M, throws errors with status code, etc included so we could share a consistent way of error handling